From 425a4e99900106228eb5268deadb4c215578bd57 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Mon, 24 Apr 2017 20:27:23 -0400
Subject: [PATCH] Add ChatColor.getById

Bukkit has had a map of this for years and it was totally unused...
---
 src/main/java/org/bukkit/ChatColor.java | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/src/main/java/org/bukkit/ChatColor.java b/src/main/java/org/bukkit/ChatColor.java
index adbae51a7..a5d53def2 100644
--- a/src/main/java/org/bukkit/ChatColor.java
+++ b/src/main/java/org/bukkit/ChatColor.java
@@ -240,6 +240,15 @@ public enum ChatColor{
     };
 
     /**
+     * Gets the numeric ID associated with this color
+     *
+     * @return An int value of this color code
+     */
+    public int getId() {
+        return intCode;
+    }
+
+    /**
      * Gets the char value associated with this color
      *
      * @return A char value of this color code
@@ -272,6 +281,17 @@ public enum ChatColor{
     }
 
     /**
+     * Gets the color represented by the specified color ID
+     *
+     * @param id Code to check
+     * @return Associative {@link org.bukkit.ChatColor} with the given id,
+     *     or null if it doesn't exist
+     */
+    public static ChatColor getById(int id) {
+        return BY_ID.get(id);
+    }
+
+    /**
      * Gets the color represented by the specified color code
      *
      * @param code Code to check
-- 
2.11.0

